prow/scripts/cluster-integration/kyma-gke-nightly.sh

```diff
@@ -0,0 +1,208 @@
+	"${KYMA_SCRIPTS_DIR}"/is-installed.sh --timeout 30m
+}
+
+function installStabilityChecker() {
+	STATS_FAILING_TEST_REGEXP=${STATS_FAILING_TEST_REGEXP:-"'\"'([0-9A-Za-z_-]+)'\"' (?:has Failed status?|failed due to too long Running status?|failed due to too long Pending status?|failed with Unknown status?)"}
+	STATS_SUCCESSFUL_TEST_REGEXP=${STATS_SUCCESSFUL_TEST_REGEXP:-"Test of '\"'([0-9A-Za-z_-]+)'\"' was successful"}
+	STATS_ENABLED=true
+	
+	SC_DIR=${KYMA_SOURCES_DIR}/tools/stability-checker
+
+	kubectl create -f "${SC_DIR}/local/provisioning.yaml"
+	bash "${SC_DIR}/local/helpers/isready.sh" kyma-system app  stability-test-provisioner
+	kubectl exec stability-test-provisioner -n kyma-system --  mkdir -p /home/input
+	kubectl cp "${KYMA_SCRIPTS_DIR}/testing.sh" stability-test-provisioner:/home/input/ -n kyma-system
+	kubectl cp "${KYMA_SCRIPTS_DIR}/utils.sh" stability-test-provisioner:/home/input/ -n kyma-system
+	kubectl delete pod -n kyma-system stability-test-provisioner
+
+	helm install --set slackClientWebhookUrl="${SLACK_CLIENT_WEBHOOK_URL}" \
+		--set slackClientChannelId="${SLACK_CLIENT_CHANNEL_ID}" \
+		--set slackClientToken="${SLACK_CLIENT_TOKEN}" \
+		--set stats.enabled="${STATS_ENABLED}" \
+		--set stats.failingTestRegexp="${STATS_FAILING_TEST_REGEXP}" \
+		--set stats.successfulTestRegexp="${STATS_SUCCESSFUL_TEST_REGEXP}" \
+		"${SC_DIR}/deploy/chart/stability-checker" \
+		--namespace=kyma-system \
+		--name=stability-checker
+}
+
+# Enforce lowercase
+readonly REPO_OWNER=$(echo "${REPO_OWNER}" | tr '[:upper:]' '[:lower:]')
+export REPO_OWNER
+readonly REPO_NAME=$(echo "${REPO_NAME}" | tr '[:upper:]' '[:lower:]')
+export REPO_NAME
+
+export TEST_INFRA_SOURCES_DIR="${KYMA_PROJECT_DIR}/test-infra"
+export KYMA_SOURCES_DIR="${KYMA_PROJECT_DIR}/kyma"
+export KYMA_SCRIPTS_DIR="${KYMA_SOURCES_DIR}/installation/scripts"
+
+readonly CURRENT_TIMESTAMP=$(date +%Y%m%d)
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+	readonly LAST_TIMESTAMP=$(date -v -1d '+%Y%m%d')
+else
+	readonly LAST_TIMESTAMP=$(date +%Y%m%d --date="yesterday")
+fi
+
+readonly COMMON_NAME=$(echo "gkeint-nightly-${CURRENT_TIMESTAMP}" | tr "[:upper:]" "[:lower:]")
+readonly KYMA_INSTALLER_IMAGE="${DOCKER_PUSH_REPOSITORY}${DOCKER_PUSH_DIRECTORY}/gke-nightly/${REPO_OWNER}/${REPO_NAME}:${CURRENT_TIMESTAMP}"
+export KYMA_INSTALLER_IMAGE
+
+### Cluster name must be less than 40 characters!
+export CLUSTER_NAME="${COMMON_NAME}"
+
+### For provision-gke-cluster.sh
+export GCLOUD_PROJECT_NAME="${CLOUDSDK_CORE_PROJECT}"
+export GCLOUD_COMPUTE_ZONE="${CLOUDSDK_COMPUTE_ZONE}"
+
+# shellcheck disable=SC1090
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/library.sh"
+
+shout "Authenticate"
+date
+init
+readonly DNS_DOMAIN="$(gcloud dns managed-zones describe "${CLOUDSDK_DNS_ZONE_NAME}" --format="value(dnsName)")"
+export DNS_DOMAIN
+
+shout "Delete old cluster - gkeint-nightly-${LAST_TIMESTAMP}"
+date
+removeCluster "${LAST_TIMESTAMP}"
```

Comment:
can this script handle situation that old cluster does not exist?

Reply:
Implementing guard to delete clusters only if we detect them